oligo_designer_toolsuite/IO/_database.py

```diff
@@ -112,104 +118,278 @@ def read_reference_DB(self, file_reference_DB):
         else:
             raise ValueError("Database file does not exist!")
 
-    def read_oligos_DB(self, file_oligos_DB):
+    def read_oligos_DB_gtf(self, file_oligos_DB_gtf, file_oligos_DB_fasta):
+        """Create the oligo db dictionary from a gtf file.
+
+        :param file_oligos_DB_gtf: Path to the file.
+        :type file_oligos_DB_gtf: str
+        :param file_oligos_DB_fasta: Path to the file.
+        :type file_oligos_DB_fasta: str
+        :raises ValueError: When the file given has the wrong format.
+        :raises ValueError: When the path given does not exist.
+        """
+
+        if os.path.exists(file_oligos_DB_gtf):
+            if data_parser.check_gtf_format(file_oligos_DB_gtf):
+                self.file_oligos_DB_gtf = file_oligos_DB_gtf
+            else:
+                raise ValueError("Database has incorrect format!")
+        else:
+            raise ValueError("Database file does not exist!")
+
+        if os.path.exists(file_oligos_DB_fasta):
+            if data_parser.check_fasta_format(file_oligos_DB_fasta):
+                self.file_oligos_DB_fasta = file_oligos_DB_fasta
+            else:
+                raise ValueError("Database has incorrect format!")
+        else:
+            raise ValueError("Database file does not exist!")
+
+        oligos_df = data_parser.read_gtf(self.file_oligos_DB_gtf)
+        oligos_fasta = SeqIO.parse(self.file_oligos_DB_fasta, "fasta")
+        self.oligos_DB = {}
+        # compute the additional columns
+        columns_fixed = [
+            "probe_sequence",
+            "transcript_id",
+            "exon_id",
+            "chromosome",
+            "start",
+            "end",
+            "strand",
+            "length",
+        ]
+        columns_df = list(oligos_df.columns)
+        # remove the columns we don't need
+        columns_df.remove("seqname")
+        columns_df.remove("source")
+        columns_df.remove("feature")
+        columns_df.remove("score")
+        columns_df.remove("frame")
+        columns_df.remove("gene_id")
+        additional_columns = [
+            column for column in columns_df if (column not in columns_fixed)
+        ]
+        # crated the oligos db
```

Comment:
@IsraMekki0 do you think it would make sense to move those parts to the _data_parser.py file?

Reply:
Yes right, it would be cleaner. And in that case I think it's better to have only one function `read_oligos_DB` with a param `format` (tsv or gtf) in `_database.py` but 2 separate functions in `_data_parser.py`. Same for `write_oligos_DB`. Does that make sense to you @francesco-campi ?